request.go

```diff
@@ -73,6 +74,20 @@ type Request struct {
 	retryConditions     []RetryConditionFunc
 }
 
+func (r *Request) GetCurlCmd() string {
```

Comment:
@ahuigo Please refactor this method with the name `GenerateCurlCommand` and update this method logic to on-demand cURL command generation.

Reply:
done